src/mds/OpenFileTable.cc

```diff
@@ -1045,10 +1045,14 @@ void OpenFileTable::_prefetch_dirfrags()
   }
 
   MDSGatherBuilder gather(g_ceph_context);
+  int num_opening_dirfrags = 0;
   for (auto dir : fetch_queue) {
     if (dir->state_test(CDir::STATE_REJOINUNDEF))
       assert(dir->get_inode()->dirfragtree.is_leaf(dir->get_frag()));
     dir->fetch(gather.new_sub());
+
+    if (!(++num_opening_dirfrags % 100))
+      mds->heartbeat_reset();
```

Comment:
Note: these don't bother me because we don't expect a non-active MDS to be responsive to clients.

Reply:
But if recovering mds does not send beacon to monitor, monitor may replace it with other mds